src/transform/FooterMenu.js

```diff
@@ -0,0 +1,218 @@
+    case MenuItemType.pageIssues:
+      return 'pagelib_footer_menu_icon_page_issues'
+    case MenuItemType.disambiguation:
+      return 'pagelib_footer_menu_icon_disambiguation'
+    case MenuItemType.coordinate:
+      return 'pagelib_footer_menu_icon_coordinate'
+    default:
+      return ''
+    }
+  }
+
+  /**
+   * Returns reference to function for extracting payload when this menu item is tapped.
+   * @return {FooterMenuItemPayloadExtractor}
+   */
+  payloadExtractor() {
+    switch (this.itemType) {
+    case MenuItemType.languages:
+      return undefined
+    case MenuItemType.lastEdited:
+      return undefined
+    case MenuItemType.pageIssues:
+      return pageIssuesStringsArray
+    case MenuItemType.disambiguation:
+      return disambiguationTitlesArray
+    case MenuItemType.coordinate:
+      return undefined
+    default:
+      return undefined
+    }
+  }
+}
+
+/**
+ * Makes document fragment for a menu item.
+ * @param {!MenuItem} menuItem
+ * @param {!Document} document
+ * @return {!DocumentFragment}
+ */
+const documentFragmentForMenuItem = (menuItem, document) => {
+  const item = document.createElement('div')
+  item.className = 'pagelib_footer_menu_item'
+
+  const containerAnchor = document.createElement('a')
+  containerAnchor.addEventListener('click', () => {
+    menuItem.clickHandler(menuItem.payload)
+  }, false)
+
+  item.appendChild(containerAnchor)
+
+  if (menuItem.title) {
+    const title = document.createElement('div')
+    title.className = 'pagelib_footer_menu_item_title'
+    title.innerText = menuItem.title
+    containerAnchor.title = menuItem.title
+    containerAnchor.appendChild(title)
+  }
+
+  if (menuItem.subtitle) {
+    const subtitle = document.createElement('div')
+    subtitle.className = 'pagelib_footer_menu_item_subtitle'
+    subtitle.innerText = menuItem.subtitle
+    containerAnchor.appendChild(subtitle)
+  }
+
+  const iconClass = menuItem.iconClass()
+  if (iconClass) {
+    item.classList.add(iconClass)
+  }
+
+  return document.createDocumentFragment().appendChild(item)
+}
+
+/**
+ * Adds a MenuItem to a container.
+ * @param {!MenuItem} menuItem
+ * @param {!string} containerID
+ * @param {!Document} document
+ */
+const addItem = (menuItem, containerID, document) => {
+  document.getElementById(containerID).appendChild(
+    documentFragmentForMenuItem(menuItem, document)
+  )
+}
+
+/**
+ * Conditionally adds a MenuItem to a container.
+ * @param {!string} title
+ * @param {!string} subtitle
+ * @param {!MenuItemType} itemType
+ * @param {!string} containerID
+ * @param {FooterMenuItemClickCallback} clickHandler
+ * @param {!Document} document
+ * @return {void}
+ */
+const maybeAddItem = (title, subtitle, itemType, containerID, clickHandler, document) => {
+  const item = new MenuItem(title, subtitle, itemType, clickHandler)
+
+  // Items are not added if they have a payload extractor which fails to extract anything.
+  if (item.payloadExtractor() !== undefined) {
```

Comment:
I think we can use the fact that a defined value will be truthy to simplify this to `if (item.payloadExtractor())`.